hostpwrctl.cpp

```diff
@@ -0,0 +1,414 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2021 YADRO.
+ */
+
+#include <sdbusplus/bus.hpp>
+#include <sdbusplus/bus/match.hpp>
+#include <sdbusplus/exception.hpp>
+#include <sdeventplus/clock.hpp>
+#include <sdeventplus/event.hpp>
+#include <sdeventplus/exception.hpp>
+#include <sdeventplus/source/event.hpp>
+#include <sdeventplus/utility/timer.hpp>
+
+#include <cstdio>
+
+constexpr auto clockId = sdeventplus::ClockId::RealTime;
+using Timer = sdeventplus::utility::Timer<clockId>;
+
+static sdbusplus::bus::bus systemBus = sdbusplus::bus::new_default();
+static sdeventplus::Event systemEvent = sdeventplus::Event::get_default();
+
+constexpr auto chassisPath = "/xyz/openbmc_project/state/chassis0";
+constexpr auto chassisIface = "xyz.openbmc_project.State.Chassis";
+constexpr auto chassisState = "CurrentPowerState";
+constexpr auto chassisStateOn =
+    "xyz.openbmc_project.State.Chassis.PowerState.On";
+constexpr auto chassisStateOff =
+    "xyz.openbmc_project.State.Chassis.PowerState.Off";
+constexpr auto chassisTransition = "RequestedPowerTransition";
+constexpr auto chassisTransitionOff =
+    "xyz.openbmc_project.State.Chassis.Transition.Off";
+
+constexpr auto hostPath = "/xyz/openbmc_project/state/host0";
+constexpr auto hostIface = "xyz.openbmc_project.State.Host";
+constexpr auto hostState = "CurrentHostState";
+constexpr auto hostStateOn = "xyz.openbmc_project.State.Host.HostState.Running";
+constexpr auto hostStateOff = "xyz.openbmc_project.State.Host.HostState.Off";
+constexpr auto hostTransition = "RequestedHostTransition";
+constexpr auto hostTransitionOn =
+    "xyz.openbmc_project.State.Host.Transition.On";
+constexpr auto hostTransitionOff =
+    "xyz.openbmc_project.State.Host.Transition.Off";
+constexpr auto hostTransitionReboot =
+    "xyz.openbmc_project.State.Host.Transition.Reboot";
+
+static std::string currentChassisState, expectedChassisState;
+static std::string currentHostState, expectedHostState;
```

Comment:
I have tried to write it quick and simple.
Yes, it might be improved, but it requires to think more about the architecture. I will do that in the following days unless something more important happens.

Reply:
ok